Problems/11051/ans_11051.py

```diff
@@ -0,0 +1,43 @@
+# https://www.acmicpc.net/problem/11051
+
+import sys
+
+
+def solve():
+    DIV = 10007
+    input = sys.stdin.readline
+    N, K = map(int, input().split())
+    # 11050 문제와 같이 재귀로 풀면 N이 큰 경우에 대해 재귀 제한에 걸려서 불가능
+    # Memory 제한조건이 256MB이므로, Bottom-up으로 주어진 Range에 대해 조합 테이블을 생성하기 충분
+    # Python은 Overflow가 존재하지 않으므로 Factorial 계산을 해버려도 무방하지만...
+    # nCk = n-1Ck + n-1Ck-1
+    nCr = {}
+    nCr[(1, 0)] = 1
+    nCr[(1, 1)] = 1
+
+    for i in range(1, N + 1):
+        for j in range(i + 1):
+            if i - 1 < 2 * j:
+                j = i - j
+            if j == 1:
+                nCr[(i, j)] = i
+            elif j == 0:
+                nCr[(i, j)] = 1
+            else:
+                # 이부분 머리아파서 정리가 안됨... 걍 빡코딩
+
+                try:
+                    t1 = nCr[(i - 1, j - 1)]
+                except:
+                    t1 = nCr[(i - 1, i - j)]
+                try:
+                    t2 = nCr[(i - 1, j)]
+                except:
+                    t2 = nCr[(i - 1, i - j - 1)]
```

Comment:
특별하게 Dictionary를 사용하는 이유가 있었는지 궁금합니다.

List를 통한 방법이 좀더 심플하지 않을까 싶어서요...

```suggestion
    # 팩토리얼보다 파스칼의 삼각형 이용(이항 계수를 삼각형 모양으로 배열한 것)
    dp = []
    for i in range(N+1):
        dp.append([1]*(i+1))
        print(dp)
    
    for i in range(2, N+1):
        for j in range(1, i):
            t1 = dp[i-1][j-1]
            t2 = dp[i-1][j]
            dp[i][j] = t1 + t2
```

Reply:
2차원배열로 만들 경우 반복문 돌리면서 쌓아나가게 될 텐데,
각 row마다 길이가 다른 리스트가 들어가는건 좋은 방법이 아니라는 생각이 들었음.
(깔끔해지려면, 정확히 사각형의 matrix가 나와야하지 않을까..?)
더불어 이 문제에선 상관없긴한데, 로직 돌아가면서 1씩 길이가 증가하는 것이 아니라,
갑자기 jump up 해서 memoization을 해야한다면 (4C2 까지밖에 적어놓은것이 없는데, 6C4 같은게 요구된다면)
리스트의 index = 이항계수 라는 공식이 깨져버리게 되는 위험도 있어보임.